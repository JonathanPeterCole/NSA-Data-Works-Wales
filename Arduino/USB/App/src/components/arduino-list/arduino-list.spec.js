import React from 'react'
import { shallow } from 'enzyme'

// Component
import ArduinoList from './arduino-list'

// Mock Arduino Connect
jest.mock('../../library/arduino-connect/arduino-connect')

// Prepare an array of test arduinos
const testArduinos = [
  {
    comName: 'COM1',
    serialNumber: '0000000001'
  },
  {
    comName: 'COM2',
    serialNumber: '0000000002'
  },
  {
    comName: 'COM3',
    serialNumber: '0000000003'
  }
]

// Tests
describe('Arduino List Component', () => {
  it('Renders without crashing', () => {
    // Attempt to render the Arduino List component
    // Shallow render to ensure we are only testing the ArduinoList component
    shallow(
      <ArduinoList
        arduinos={testArduinos}
        disconnected={null}
        api={null} />
    )
  })

  it('Matches snapshot', () => {
    // Compare the component to the snapshot
    const wrapper = shallow(
      <ArduinoList
        arduinos={testArduinos}
        disconnected={null}
        api={null} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Lists correct number of Arduinos', () => {
    const wrapper = shallow(
      <ArduinoList
        arduinos={testArduinos}
        disconnected={null}
        api={null} />
    )
    expect(wrapper.find('Arduino')).toHaveLength(3)
  })
})
