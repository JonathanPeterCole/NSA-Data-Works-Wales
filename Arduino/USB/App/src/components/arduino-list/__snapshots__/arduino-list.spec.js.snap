// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arduino List Component Matches snapshot 1`] = `
<div
  className="arduino-list"
>
  <div
    className="scroll-fader"
  />
  <div
    className="scroll-area"
    onScroll={[Function]}
  >
    <TransitionGroup
      childFactory={[Function]}
      className="arduino-container"
      component="div"
    >
      <CSSTransition
        classNames="arduino-transition"
        key="0"
        timeout={300}
        unmountOnExit={true}
      >
        <Arduino
          api={null}
          arduino={
            Object {
              "comName": "COM1",
              "serialNumber": "0000000001",
            }
          }
          disconnected={null}
          key="0"
        />
      </CSSTransition>
      <CSSTransition
        classNames="arduino-transition"
        key="1"
        timeout={300}
        unmountOnExit={true}
      >
        <Arduino
          api={null}
          arduino={
            Object {
              "comName": "COM2",
              "serialNumber": "0000000002",
            }
          }
          disconnected={null}
          key="1"
        />
      </CSSTransition>
      <CSSTransition
        classNames="arduino-transition"
        key="2"
        timeout={300}
        unmountOnExit={true}
      >
        <Arduino
          api={null}
          arduino={
            Object {
              "comName": "COM3",
              "serialNumber": "0000000003",
            }
          }
          disconnected={null}
          key="2"
        />
      </CSSTransition>
    </TransitionGroup>
  </div>
</div>
`;
